src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -435,6 +435,32 @@ namespace winrt::TerminalApp::implementation
     void CommandPalette::_rootPointerPressed(Windows::Foundation::IInspectable const& /*sender*/,
                                              Windows::UI::Xaml::Input::PointerRoutedEventArgs const& /*e*/)
     {
+        if (Visibility() != Visibility::Collapsed)
+        {
+            _dismissPalette();
+        }
+    }
+
+    // Method Description:
+    // - The purpose of this event handler is to hide the palette if it loses focus.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void CommandPalette::_lostFocusHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                           Windows::UI::Xaml::RoutedEventArgs const& /*args*/)
+    {
+        auto focusedElement = Input::FocusManager::GetFocusedElement(this->XamlRoot()).try_as<DependencyObject>();
+        while (focusedElement)
+        {
+            if (focusedElement == *this)
+            {
+                return;
+            }
+
+            focusedElement = winrt::Windows::UI::Xaml::Media::VisualTreeHelper::GetParent(focusedElement);
```

Comment:
@DHowett - I mean you were right, the even is getting routed to us if one of our children loses focus. We only need to distinguish if it loses the focus in favor of some other our child or some external element.

We cannot achieve this with the `sender`, because it is always the element that loses focus, while we are interested in element that gets focus.

Reply:
I guess I will issue another commit explaining this :blush: